### ISSUE TITLE
Unable to pass "Predicate" argument to ``wait for`` keywords

### ISSUE THREAD

**Issue by Leemur89** [MarketSquare/robotframework-browser#1486]
**Describe the bug**
I cannot pass in input of ``Wait For Response`` a predicate matcher, it is always considered as a url.
Documentation states

> matcher Request URL string, JavaScript regex or JavaScript function to match request by. By default (with empty string) matches first available request.

But there are no example on how to pass a predicate

**To Reproduce**
```
Wait For Response  matcher=request => request.url() === 'https://example.com' && request.method() === 'GET'
```

**Expected behavior**
Wait for the request with given url and method

**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser: Chromium
 - Version: robotframework-browser==9.0.2

**Additional context**
Note that my need behind is to be able to wait for a specific request method (sometimes I am polluted with the OPTIONS requests, so I want to filter them out)

**Comment by aaltat** [comment]
I recall seeing this bug in the node side, it did had something to do with how we handle the arguments.

**Comment by aaltat** [comment]
@allcontributors please add @Leemur89 for bugs.

**Comment by jpobley** [comment]
**Desktop (please complete the following information):**
OS: MacOS 11.6
Browser: Chromium
Version: robotframework-browser==10.0.3


I also ran into this issue, but with a `POST` request:

**Test Data**
```
${response}    Wait For Response    (resp) => resp.url().includes('jsonplaceholder');
```
**Log Output**

```
TimeoutError: page.waitForResponse: Timeout while waiting for event "response"
=========================== logs ===========================
waiting for response /.*(resp) => resp.url().includes('jsonplaceholder')…/
============================================================
```

Next, if I use a simple string for the `matcher`, then I get a different error:

**Test Data**
```
${res}    Wait For Response    matcher=jsonplaceholder
```

**Log Output**
```
Error: response.text: Protocol error (Network.getResponseBody): No resource with given identifier found
```

Using a string with a `GET` request works as expected.

Curiously, `waitForRequest` and `waitForResponse` do different things; I'm not sure if it's related:

[waitForResponse](https://github.com/MarketSquare/robotframework-browser/blob/main/node/playwright-wrapper/network.ts#L54
)
```js
export async function waitForResponse(request: pb.Request.HttpCapture, page: Page): Promise<pb.Response.Json> {
    const urlOrPredicate = new RegExp(`.*${request.getUrlorpredicate()}`);
    ...
    ...
}
```

[waitForRequst](https://github.com/MarketSquare/robotframework-browser/blob/main/node/playwright-wrapper/network.ts#L75)
```js
export async function waitForRequest(request: pb.Request.HttpCapture, page: Page): Promise<pb.Response.String> {
    const urlOrPredicate = request.getUrlorpredicate();
    ...
    ...
}
````

**Comment by jpobley** [comment]
I patched this locally, and it worked as expected. I can't seem to get `inv atest` or `inv utest` to completely pass, though, and I had to modify `tasks.py` to build the node dependencies for the Apple M1 chip , so I'm reluctant to open a PR.

```typescript
export function deserializeUrlOrPredicate(request: pb.Request.HttpCapture): RegExp | Function {
    let urlOrPredicate = request.getUrlorpredicate();

    if (/^function.*{.*}$/.test(urlOrPredicate)) {
        // eval() needs an expression in order to return a function,
        // so if it's not an arrow expression, it must be wrapped in parens
        urlOrPredicate = `(${urlOrPredicate})`;
    }

    // Possibly an arrow function or a parens-wrapped function from above
    if (urlOrPredicate.startsWith('(')) {
        const fn = eval(urlOrPredicate);
        if (typeof fn === 'function' || Object.prototype.toString.call(fn) === '[object Function]') {
            return fn;
        }
    }

    return new RegExp(`.*${urlOrPredicate}`);
}
```

**Comment by aaltat** [comment]
1) Please send a PR, we have few rand

**Comment by rsandbach** [comment]
This should be resolved with PR#1533. I wasn't sure how to associate it with this issue.
